client/src/handlers/handlers.ts

```diff
@@ -14,6 +19,211 @@ export function makeHandleRunJob (
+    passCommandToTerminal(cmd, terminal)  
+}
+
+/**
+ * Run main without any custom arguments
+ * 
+ * Sends command `java -jar <path_to_flix.jar> <paths_to_all_flix_files>` to an existing (if already exists else new) terminal.
+ * 
+ * @param context vscode.ExtensionContext
+ * 
+ * @param launchOptions LaunchOptions
+ * 
+ * @return function handler
+*/
+export function cmdRunMain(
+    context: vscode.ExtensionContext, 
+    launchOptions: LaunchOptions = defaultLaunchOptions
+    ) {
+        return async function handler () {
+            const globalStoragePath = context.globalStoragePath
+            const workspaceFolders = _.map(_.flow(_.get('uri'), _.get('fsPath')), vscode.workspace.workspaceFolders)
+            const flixFileLocation = await ensureFlixExists({ globalStoragePath, workspaceFolders, shouldUpdateFlix: launchOptions.shouldUpdateFlix })
+            var flixFilename = quote([flixFileLocation])
+            let cmd = "java -jar "+ flixFilename + " "
+            cmd += await getFiles()
+            let terminal = ensureFlixTerminal()
+            passCommandToTerminal(cmd, terminal)  
+        }
+}
+
+/**
+ * Run main with user provided arguments
+ * 
+ * Sends command `java -jar <path_to_flix.jar> <paths_to_all_flix_files> --args <arguments>` to an existing (if already exists else new) terminal.
+ * 
+ * @param context vscode.ExtensionContext
+ * 
+ * @param launchOptions LaunchOptions
+ * 
+ * @return function handler
+*/
+export function runMainWithArgs(
+    context: vscode.ExtensionContext, 
+    launchOptions: LaunchOptions = defaultLaunchOptions
+    ) {
+        return async function handler () {
+            const globalStoragePath = context.globalStoragePath
+            const workspaceFolders = _.map(_.flow(_.get('uri'), _.get('fsPath')), vscode.workspace.workspaceFolders)
+            const flixFileLocation = await ensureFlixExists({ globalStoragePath, workspaceFolders, shouldUpdateFlix: launchOptions.shouldUpdateFlix })
+            var flixFilename = quote([flixFileLocation])
+            
+            let terminal = ensureFlixTerminal()
+            await passArgs(terminal, flixFilename)
+        }
+}
+
+/**
+ * Run main without any custom arguments in a new terminal
+ * 
+ * Sends command `java -jar <path_to_flix.jar> <paths_to_all_flix_files>` to a new terminal.
+ * 
+ * @param context vscode.ExtensionContext
+ * 
+ * @param launchOptions LaunchOptions
+ * 
+ * @return function handler
+*/
+export function runMainNewTerminal(
+    context: vscode.ExtensionContext, 
+    launchOptions: LaunchOptions = defaultLaunchOptions
+    ) {
+        return async function handler () {
+            const globalStoragePath = context.globalStoragePath
+            const workspaceFolders = _.map(_.flow(_.get('uri'), _.get('fsPath')), vscode.workspace.workspaceFolders)
+            const flixFileLocation = await ensureFlixExists({ globalStoragePath, workspaceFolders, shouldUpdateFlix: launchOptions.shouldUpdateFlix })
```

Comment:
The above three lines seems to be copied in several functions. Create a helper function?